compiler-plugin/src/main/java/io/ballerina/stdlib/email/compiler/EmailServiceValidator.java

```diff
@@ -276,6 +279,35 @@ private void validateParameter(SeparatedNodeList<ParameterNode> parameterNodes,
         }
     }
 
+    private boolean validateOnMessageParams(FunctionDefinitionNode functionDefinitionNode) {
+        FunctionTypeSymbol functionTypeSymbol = ((MethodSymbol) ctx.semanticModel().symbol(functionDefinitionNode)
+                .get()).typeDescriptor();
+        List<ParameterSymbol> inputParams;
+        if (functionTypeSymbol.params().isPresent()) {
+            inputParams = functionTypeSymbol.params().get();
+            if (inputParams.size() != 1) {
+                return false;
+            }
+            String moduleId = getModuleId(inputParams.get(0));
+            String typeDescriptorSignature = inputParams.get(0).typeDescriptor().signature();
+            return typeDescriptorSignature.equals(moduleId + ":" + EMAIL_MESSAGE);
```

Comment:
What exactly are we trying to validate here?

Seems like we are just checking whether the parameter type name is `EMAIL_MESSAGE` constant value. 

Because here we are taking the `typeDescriptor` of the `inputParams[0]`, and then taking the `moduleId` of that parameter from another way, and just compare whether they are equal?

Reply:
To explain this further, as it seems, here we have to verify whether we are passing the `email:Message` (Assuming the value of the constant `EMAIL_MESSAGE` is `Message`) type as the first parameter.

In that case, we have to check whether the module id is `email`, and the type name is `Message`. Here what you are doing is comparing the module ID of the parameter with the module ID of that parameter (which will be always true).

This check fails only when the type name of the parameter is different from `Message`. I guess that's not the intension here.

Reply:
Changed as given in your code.

Reply:
I still don't get the idea. Is any parameter type named `Message` allowed as the input parameter? Don't we have to limit this to the `email` module specific type?

Reply:
Added validation to `email` package as well.